dockstore-integration-testing/src/test/java/io/dockstore/webservice/TokenResourceIT.java

```diff
@@ -256,6 +258,158 @@ public void testNinjaedGitHubUser() throws Exception {
+        Assert.assertEquals(GOOGLE_ACCOUNT_USERNAME2, otherGoogleUserToken.getUsername());
+    }
+
+    /**
+     *
+     * @throws Exception
+     */
+    private void registerNewUsersWithExisting(TokensApi unAuthenticatedTokensApi) throws Exception {
+        mockGitHub(CUSTOM_USERNAME1);
+        mockGoogleHelper(GOOGLE_ACCOUNT_USERNAME1);
+        // Cannot create new user with the same Google account
+        try {
+            unAuthenticatedTokensApi.addGoogleToken(satellizerJSONForRegistration);
+            Assert.fail();
+        } catch (Exception e){
+            Assert.assertEquals("User already exists, cannot register new user", e.getMessage());;
+            // Call should fail
+        }
+
+        // Cannot create new user with the same GitHub account
+        try {
+            unAuthenticatedTokensApi.addToken(satellizerJSONForRegistration);
+            Assert.fail();
+        } catch (Exception e){
+            Assert.assertTrue(e.getMessage().contains("already exists"));
+            // Call should fail
+        }
+    }
+
+    /**
+     * After self-destructing the GOOGLE_ACCOUNT_USERNAME1, its previous linked accounts can be used:
+     * GOOGLE_ACCOUNT_USERNAME1 Google account and CUSTOM_USERNAME1 GitHub account
+     * @throws Exception
+     */
+    private void registerNewUsersAfterSelfDestruct(TokensApi unAuthenticatedTokensApi) throws Exception {
+        UsersApi mainUsersApi = new UsersApi(getWebClient(true, GOOGLE_ACCOUNT_USERNAME1));
+        Boolean aBoolean = mainUsersApi.selfDestruct();
+        assertTrue(aBoolean);
+        mockGoogleHelper(GOOGLE_ACCOUNT_USERNAME1);
+        io.swagger.client.model.Token recreatedGoogleToken = unAuthenticatedTokensApi
+                .addGoogleToken(satellizerJSONForRegistration);
+        mockGitHub(CUSTOM_USERNAME1);
+        io.swagger.client.model.Token recreatedGitHubToken = unAuthenticatedTokensApi
+                .addToken(satellizerJSONForRegistration);
+        assertNotSame(recreatedGitHubToken.getUserId(), recreatedGoogleToken.getUserId());
+    }
+
+    /**
+     * Dockstore account 1: has GOOGLE_ACCOUNT_USERNAME1 Google account linked
+     * Dockstore account 2: has GITHUB_ACCOUNT_USERNAME GitHub account linked
+     * Trying to link GOOGLE_ACCOUNT_USERNAME1 Google account to Dockstore account 2 should fail
+     * @throws Exception
+     */
+    private void addUnavailableGoogleTokenToGitHubUser() throws Exception {
+        mockGoogleHelper(GOOGLE_ACCOUNT_USERNAME1);
+        TokensApi otherUserTokensApi = new TokensApi(getWebClient(true, GITHUB_ACCOUNT_USERNAME));
+        // Cannot add token to other user with the same Google account
+        try {
+            otherUserTokensApi.addGoogleToken(satellizerJSON);
+            Assert.fail();
+        } catch (Exception e){
+            Assert.assertTrue(e.getMessage().contains("already exists."));;
```

Comment:
Leave out the punctuation at minimum
Test based on HTTP code would be better. The message might easily change if for example we change it to something like "already exists, you have probably registered"